lib/async/http/protocol/http1.rb

```diff
@@ -27,7 +28,7 @@ def self.client(peer)
 				end
 				
 				def self.server(peer)
-					stream = IO::Stream.new(peer, sync: true)
+					stream = peer.is_a?(IO::Stream) ? peer : IO::Stream.new(peer, sync: true)
```

Comment:
I need to think a little more about this.

Eventally, `Async::IO` is going away. I'm not sure if `Async::IO::Stream` remains or not - probably not. This was for compatibility with shims/Ruby < 3.1.

Reply:
This doesn't add a new dependency on `Async::IO::Stream`, but merely allows for `peer` to have been turned into a `Stream` one call earlier in the stack, which is needed for the new h1 vs h2 detection to function. 

If all of `Stream` is just a shim (or ends up that way in a newer Ruby), pulling it out later should be no problem. If `Stream` is adding additional functionality over standard Ruby, then there's a bit more to consider, but that feels beyond the scope of this PR.